src/components/Toggle/__tests__/toggle-tests.jest.js

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import userEvent from "@testing-library/user-event";
+import { render, cleanup } from "@testing-library/react";
+import { expect as sinonExpect } from "../../../test/test-helpers";
+import Toggle from "../Toggle";
+
+
+describe("Toggle tests", () => {
+  const formName = "myForm";
+  const toggleRole = "switch";
+
+  describe("Default selected mode", () => {
+
+    afterEach(() => {
+      cleanup();
+    });
+
+    it("should change state to off when is default selected and clicked", () => {
+      const { getByRole } = render(
+        <form name={formName}>
+          <Toggle componentClassName="dummy-class-name" isDefaultSelected />
+        </form>
+      );
+
+      const toggle = getByRole(toggleRole);
+      userEvent.click(toggle);
+      expect(toggle.checked).toBeFalsy();
+    });
+
+    it("should change state to on when is default not selected and clicked", () => {
+      const { getByRole } = render(
+        <form name={formName}>
+          <Toggle componentClassName="dummy-class-name" isDefaultSelected={false} />
+        </form>
+      );
+      const toggle = getByRole(toggleRole);
+      userEvent.click(toggle);
+      expect(toggle.checked).toBeTruthy();
+    });
+
+    it("should not change state when disabled, default selected and clicked", () => {
+      const { getByRole } = render(
+        <form name={formName}>
+          <Toggle isDisabled={true} componentClassName="dummy-class-name" isDefaultSelected />
+        </form>
+      );
+
+      const toggle = getByRole(toggleRole);
+      userEvent.click(toggle);
+      expect(toggle.checked).toBeTruthy();
+    });
+  });
+
+  describe("a11y", () => {
+    it("should add the aria label", () => {
```

Comment:
I think we should check the existence of a11y props by snapshot tests. what do you think? @orrgottlieb

Reply:
I disagree, it depends. when you can simulate interactivity via a11y props so it will simulated screen reader interactibilty

Reply:
So let's keep the aria props tests like this